haystack/nodes/retriever/link_content.py

```diff
@@ -0,0 +1,202 @@
+import logging
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional, Dict, List, Union, Callable, Any, Tuple
+from urllib.parse import urlparse
+
+import requests
+from boilerpy3 import extractors
+from requests import Response
+from requests.exceptions import InvalidURL
+
+from haystack import __version__
+from haystack.nodes import PreProcessor, BaseComponent
+from haystack.schema import Document, MultiLabel
+
+logger = logging.getLogger(__name__)
+
+
+def html_content_handler(response: Response, raise_on_failure: bool = False) -> Optional[str]:
+    """
+    Extracts content from the response text using the boilerpy3 extractor.
+    """
+    extractor = extractors.ArticleExtractor(raise_on_failure=raise_on_failure)
+    content = ""
+    try:
+        content = extractor.get_content(response.text)
+    except Exception as e:
+        if raise_on_failure:
+            raise e
+        logger.debug("Couldn't extract content from %s", response.url)
```

Comment:
I like the new approach with `raise_on_failure`, but I would also like to have a higher logging level (info? warn?).

Reply:
I thought about this extensively. It will create panic for users with a default logging value set to a warning. Ask Tuana for stories. I highly respect your feedback here, and I would have commented the same, but I am convinced these should be on debug because a) they fail a lot, and I mean a lot, and b) expert users will know how to set the logger to understand and see failures if needed. Users who set `raise_on_failure` to True will get exceptions as expected.  The best-balanced solution IMHO. Let's ask @silvanocerza and @ZanSara for opinions.

Reply:
A bit of background: I selected debug level instead of warning because when we retrieve many links in WebRetriever, quite a few fail. They fail because websites increasingly block non-human traffic. Some parsing also fails because we use licence friendly HTML parser, which could be better, but it works. Having these failed links and parsing logged at warn looks scary to users. Tuana told me she's seeing many users complaining about these warning logs. That's why. Does it make better sense with this background info?

Reply:
I'd prefer it to be a warning or info to be fair. If you users don't want to see logging they can disable it. 🤷

Reply:
Also the logic is wrong and the log won't be ever printed as of now.

If `raise_on_failure` is `False` the extractor won't even raise any exception.
This would both log and reraise the exception, but at that point might as well just let the extractor raise and remove the `try`.
```
        logger.debug("Couldn't extract content from %s", response.url)
        raise e
```